apps/frontend/src/app/Solver/GOSolver/polyUB.ts

```diff
@@ -0,0 +1,196 @@
+        if (xs.length !== 1) throw new PolyError("Multivariate", operation)
+
+        const x = map(xOp), c = op(f.operands.filter(op => op.operation === "const")
+          .map(c => (c as ConstantNode<number>).value))
+        if ((operation === "max" && context === lower) || (operation === "min" && context === upper))
+          return x
+        const { min, max } = minMaxes.get(xOp)!, yMin = op([min, c]), yMax = op([max, c])
+        return interpolate(min, yMin, max, yMax, x, context === upper)
+      }
+      case "res": {
+        if (context !== upper) throw new PolyError("Unsupported direction", operation)
+        const op = allOperations[operation]
+        const [xOp] = f.operands, { min, max } = minMaxes.get(xOp)!
+        const x = map(xOp, oppositeContext)
+        // Linear region 1 - base/2 or concave region with peak at base = 0
+        if (min < 0 && max < 1.75) return addP(1, mulP(-.5, x))
+        // Clamp `min` to guarantee upper bound
+        else return interpolate(min, op([min]), max, op([max]), x, context === upper)
+      }
+      case "sum_frac": {
+        if (context !== upper) throw new PolyError("Unsupported direction", operation)
+        const [xOp, cOp] = f.operands
+        if (cOp.operation !== "const") throw new PolyError("Non-constant node", operation)
+        const x = map(xOp), c = cOp.value, { min, max } = minMaxes.get(xOp)!
+        const loc = Math.sqrt((min + c) * (max + c))
+        if (min <= -c) throw new PolyError("Unsupported pattern", operation)
+        return slopePoint(c / (c + loc) / (c + loc), loc, loc / (loc + c), x)
+      }
+      case "threshold": {
+        const [vOp, tOp, pOp, fOp] = f.operands
+        if (fOp.operation !== "const" || tOp.operation !== "const")
+          throw new PolyError("Non-constant node", operation)
+        if (pOp.operation !== "const") {
+          if (fOp.value !== 0) throw new PolyError("Unsupported pattern", operation)
+
+          const threshOp = threshold(vOp, tOp, 1, fOp), mulOp = prod(threshOp, pOp)
+          // Populate `minMaxes` to ensure consistency
+          const { min, max } = minMaxes.get(pOp)!
+          minMaxes.set(threshOp, { min: 0, max: 1 })
+          minMaxes.set(mulOp, { min: Math.min(min, 0), max: Math.max(max, 0) })
+          return map(mulOp)
+        }
+        const { min, max } = minMaxes.get(vOp)!
+        const thresh = tOp.value, pass = pOp.value, fail = fOp.value
+        const isFirstHalf = (pass > fail) === (context === upper)
+
+        const v = map(vOp, pass > fail ? context : oppositeContext)
+        const yThresh = isFirstHalf ? pass : fail
+        const slope = (pass - fail) / (isFirstHalf ? (thresh - min) : (max - thresh))
+        return slopePoint(slope, thresh, yThresh, v)
+      }
+      default: assertUnreachable(operation)
+    }
+  })
+
+  return poly.map(p => expandPoly(p))
+}
+
+export type SumOfMonomials = Monomial[]
+type Monomial = {
+  $k: number
```

Comment:
Can we switch to `$c` so that they are consistent with other types, or is there a reason for the difference?